src/Terminal/commands/rm.ts

```diff
@@ -39,7 +37,7 @@ export function rm(args: (string | number | boolean)[], server: BaseServer): voi
   const files: FilePath[] = [];
 
   for (const target of targets) {
-    if (!hasTextExtension(target) && !hasScriptExtension(target)) {
+    if (isDirectoryPath(target) || !/\..+$/.test(target)) {
```

Comment:
This won't work.

There is *no* way to tell whether or not something is a directory, just by looking at its filename. For instance, the path `/foo.txt/blar.txt` is valid, so `/foo.txt` could be a file or a directory. So you can only tell that a path is a directory from it being a validly-constructed path (`Terminal.getDirectory` returns non-null) *and* it's not itself a file *and* it is a prefix of some existing files.

Reply:
I'm not quite sure how to properly handle the directory names, since the file-system is one big mess.
During my testing I found out that there can be directories and files with the same name.
I tried to avoid the problem here and did the following:
- If the target has no extension, it is a directory. That's the check for a match against the regex.
- Else, the function `isDirectoryPath` handles how the input should be properly formatted. If there is `a.js` as a file and a directory, the input `a.js/` will reference the directory and not the file.